src/hotspot/share/services/diagnosticCommand.cpp

```diff
@@ -472,7 +472,7 @@ void FinalizerInfoDCmd::execute(DCmdSource source, TRAPS) {
 #if INCLUDE_SERVICES // Heap dumping/inspection supported
 HeapDumpDCmd::HeapDumpDCmd(outputStream* output, bool heap) :
                            DCmdWithParser(output, heap),
-  _filename("filename","Name of the dump file", "STRING", false, "-XX:HeapDumpPath"),
+  _filename("filename","Name of the dump file", "STRING", false, "if no filename was specified, but -XX:HeapDumpPath=hdp is set, path hdp is taken"),
```

Comment:
This seems clumsy, but I'm having a hard time coming up with something better.

"the filename specified by -XX:HeapDumpPath, if specified"
"If -XX:HeapDumpPath is specified, then it is used as the default"

Makes me wonder if this approach is wrong since it is hard to document clearly. Maybe we should go back to the original fix which did not impact the jcmd. I understand the argument for having the jcmd use the HeapDumpPath setting, but it might not be worth the documentation confusion it introduces. You can argue that HeapDumpPath really is just intended to help support HeapDumpOnOutOfMemoryError. Does the jcmd also use 
HeapDumpGzipLevel? I'm not sure, but we should be consistent with the application of HeapDumpPath and HeapDumpGzipLevel to the jcmd.

Reply:
So far we only use the gzip level setting from jcmd, not HeapDumpGzipLevel .
See the `level` variable in  `HeapDumpDCmd::execute` . Yeah you are probably right we should make it consistent.

Reply:
> If -XX:HeapDumpPath is specified, then it is used as the default

No, the filename set with jcmd GC:heamp_dump  filename   has priority. So we should better keep the current description.

Reply:
So should I also use  HeapDumpGzipLevel  the same way as HeapDumpPath ? Tehn we have to change the text in globals.hpp for HeapDumpGzipLevel as well because it mentions only the HeapDumpOnOutOfmemoryError case and not the jcmd case .

Reply:
I actually mentioned  HeapDumpGzipLevel as another reason not to make this change. I'm still not seeing it's value relative to the documentation headaches it causes.

@ansteiner commented that:

> This is really helpful from a support point of view.

I'd like to understand how. It seems to me that the person using the jcmd has a lot more interest in where the file ends up than the person launching the JVM, and can always specify the location with the jcmd. It seems odd to me that the jcmd user would want to rely on HeapDumpPath when they can just specify the path with the jcmd and know for sure where the file is going to end up.

Reply:
In a cloud environment using containers, the HeapDumpPath is automatically set to a file system service to persist the heapdump. However, if a support engineer or DevOps detects high or increasing memory usage and wants to trigger a heapdump via jcmd, they would need to specify the filename. This requires retrieving the set HeapDumpPath from the app environment and copying it to the jcmd to use the persistent file system. This change can help avoid the need for an additional copy and paste step, which is prone to errors.